google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/git/UploadSourceAction.java

```diff
@@ -468,9 +469,12 @@ public void run() {
         return false;
       }
 
-      Collection<VirtualFile> files2add =
-          CollectionUtils.intersection(untrackedFiles, files2commit);
-      Collection<VirtualFile> files2rm = ContainerUtil.subtract(trackedFiles, files2commit);
+      Set<VirtualFile> files2CommitAsSet = new HashSet<>(files2commit);
+      Set<VirtualFile> untrackedFilesAsSet = new HashSet<>(untrackedFiles);
```

Comment:
this looks a lot better and is more explicit. 

When I look at this performFirstCommitIfRequired method I'm a bit surprised that we have to define this kind of thing. It feels very standard git workflowy. But I don't want to get into changing that.  My main/only concern is that the behavior remains the same for now. It would be cool to have some tests to reflect that.